### ISSUE TITLE
Update Monorepo to PNPM v7

### ISSUE THREAD

**Issue by ObliviousHarmony** [woothemes/woocommerce#33686]
### Describe the solution you'd like

Update the version of PNPM in the monorepo to v7.

Closes https://github.com/woocommerce/woocommerce/issues/32885.

#### Prerequisites

- [x] https://github.com/woocommerce/woocommerce/issues/33683
- [x] https://github.com/woocommerce/woocommerce/issues/33684
- [x] https://github.com/woocommerce/woocommerce/issues/33685
- [x] https://github.com/woocommerce/woocommerce/issues/34203

**Comment by psealock** [comment]
Unblocked because we are at the point of being able to try again and see whats broken.

**Comment by jonathansadowski** [comment]
I made some progress in this draft PR:

* https://github.com/woocommerce/woocommerce/pull/34536

but I ran into typescript issues related to `oclif` in a few of our tools that I wasn't quite able to resolve (they show up in most of the CI... and also if you just start fresh and do a `pnpm i` with pnpm 7 installed globally). Maybe somebody who understands typescript better than I can take a look.

I'm going to unassign myself from this, as I'm out for the next week, in case somebody else wants to pick up.

Some notes:
* I ran into a bunch of issues with existing caches / node modules / etc, and found it was easiest to just start with a fresh checkout in a new directory, and then upgrade to pnpm 7 by doing `npm i -g pnpm`. (The method I use for downgrading is `npm uninstall -g pnpm && npm i -g pnpm@6.24.2`).
* I found some of the notes on [this guide](https://www.raulmelo.dev/blog/migrating-from-pnpm-6-to-7) helpful, especially the bit about running a script, and the filter command now needing to be at the beginning.
